src/app/@ansyn/imagery/stay-in-imagery-service/stay-in-imagery.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Inject, Injectable } from '@angular/core';
+import { IImageryConfig } from '../model/iimagery-config';
+import { IMAGERY_CONFIG } from '../model/configuration.token';
+
+@Injectable()
+export class StayInImageryService {
+	elementCallback: Function;
+	timerCallback: Function;
+	timerId: number;
+	verticalPadding = this.imageryConfig.stayInImageryVerticalPadding || 0;
+	public moveLeft = 0;
+	public moveDown = 0;
+
+	constructor(@Inject(IMAGERY_CONFIG) protected imageryConfig: IImageryConfig) {
+	}
+
+	init(elementOrElementCallback: Function | Element, timerCallback: Function = null) {
+		this.elementCallback = elementOrElementCallback instanceof Function ? elementOrElementCallback : () => elementOrElementCallback;
+		this.timerCallback = timerCallback;
+		this.timerId = window.setInterval(this.calcPositionToStayInsideImagery.bind(this), 300);
+	}
+
+	destroy() {
```

Comment:
check if there is a build in ngOnDestroy for service

Reply:
1 Yes there is ngOnDestroy for service!
2 I added it to the Stay service, and deleted the destroy() method.